server/controllers/refer/referralsController.test.ts

```diff
@@ -318,7 +318,28 @@ describe('ReferralsController', () => {
         await requestHandler(request, response, next)
 
         expect(response.redirect).toHaveBeenCalledWith(referPaths.show({ referralId: referral.id }))
-        expect(referralService.updateReferral).toHaveBeenCalledWith(token, referral.id, { oasysConfirmed: true })
+        expect(referralService.updateReferral).toHaveBeenCalledWith(token, referral.id, {
+          oasysConfirmed: true,
+          reason: referral.reason,
+        })
+      })
+    })
+
+    describe('updating `reason`', () => {
+      it('asks the service to update the field and redirects to the show action', async () => {
+        const referral = referralFactory.build({ oasysConfirmed: true, reason: undefined })
+        referralService.getReferral.mockResolvedValue(referral)
+
+        request.body.reason = ' Some reason\nAnother paragraph\n '
+
+        const requestHandler = referralsController.update()
+        await requestHandler(request, response, next)
+
+        expect(response.redirect).toHaveBeenCalledWith(referPaths.show({ referralId: referral.id }))
+        expect(referralService.updateReferral).toHaveBeenCalledWith(token, referral.id, {
+          oasysConfirmed: true,
+          reason: 'Some reason\nAnother paragraph',
+        })
```

Comment:
Shall we do the same as with `oasysConfirmed` here - not set `oasysConfirmed` when building the referral, but check that the current value is sent to `updateReferral`?

Reply:
Think this has highlighted an issue with setting `request` values in each of the tests without resetting.  Will investigate.